From bdad3016d49579a850af09424dcacb9554a38217 Mon Sep 17 00:00:00 2001
From: Limeng <Meng.Li@windriver.com>
Date: Mon, 8 Jun 2020 16:27:45 +0800
Subject: [PATCH 4/9] driver: media: imx8: clear endpoint struct variable
 before using it

Clear endpoint struct variable before using it, otherwise code
enters wrong case.

Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/media/platform/imx8/mxc-media-dev.c | 1 +
 drivers/media/platform/imx8/mxc-mipi-csi2.c | 1 +
 2 files changed, 2 insertions(+)

diff --git a/drivers/media/platform/imx8/mxc-media-dev.c b/drivers/media/platform/imx8/mxc-media-dev.c
index 0d83fc4eced7..b0170f519791 100644
--- a/drivers/media/platform/imx8/mxc-media-dev.c
+++ b/drivers/media/platform/imx8/mxc-media-dev.c
@@ -424,6 +424,7 @@ static int register_sensor_entities(struct mxc_md *mxc_md)
 		if (!ep)
 			return -EINVAL;
 
+		memset(&endpoint, 0, sizeof(endpoint));
 		v4l2_fwnode_endpoint_parse(of_fwnode_handle(ep), &endpoint);
 		if (WARN_ON(endpoint.base.port >= MXC_MAX_SENSORS)) {
 			v4l2_err(&mxc_md->v4l2_dev, "Failed to get sensor endpoint\n");
diff --git a/drivers/media/platform/imx8/mxc-mipi-csi2.c b/drivers/media/platform/imx8/mxc-mipi-csi2.c
index aaa0e638800a..cf9ce06c44aa 100644
--- a/drivers/media/platform/imx8/mxc-mipi-csi2.c
+++ b/drivers/media/platform/imx8/mxc-mipi-csi2.c
@@ -675,6 +675,7 @@ static int mipi_csi2_parse_dt(struct mxc_mipi_csi2_dev *csi2dev)
 	}
 
 	/* Get port node */
+	memset(&endpoint, 0x0, sizeof(endpoint));
 	v4l2_fwnode_endpoint_parse(of_fwnode_handle(ep), &endpoint);
 
 	csi2dev->num_lanes = endpoint.bus.mipi_csi2.num_data_lanes;
-- 
2.17.1

